x-pack/metricbeat/module/gcp/billing/billing.go

```diff
@@ -302,10 +367,10 @@ func getCurrentDate() string {
 	return fmt.Sprintf("%04d%02d%02d", currentTime.Year(), int(currentTime.Month()), currentTime.Day())
 }
 
-func generateEventID(currentDate string, rowItems []bigquery.Value) string {
+func generateEventID(currentDate string, row row) string {
 	// create eventID using hash of current_date + invoice.month + project.id + project.name
 	// This will prevent more than one billing metric getting collected in the same day.
-	eventID := currentDate + rowItems[0].(string) + rowItems[1].(string) + rowItems[2].(string)
+	eventID := currentDate + row.InvoiceMonth + row.ProjectId + row.ProjectName
```

Comment:
My concern with this implementation is that we no longer have the original event which aggregates `total_exact`. Instead, we now have all of the events that add up to `total_exact`.

There is also an issue where the ID of the event is always the same in `generateEventID` for all documents, resulting in only one document.

I think we should have the events grouped by sku/service as well as the original event that is currently used in dashboards. WDYT @zmoog?

Reply:
@kaiyan-sheng, IIRC, you worked on this metricset in the past. I would love to read your comments when you have time.

@gpop63, do we also need to update the dashboard?

Reply:
@zmoog We'll have access to more granular data like `sku_id`, `service_id`, and `effective_price`, which will allow us to create different lenses for breaking down the data by these fields. However, given that we previously had only 1 document per day, the aggregation functions in current lenses must change from `max` to `sum`.

I'm also considering how to best manage the `tags` field so they might be used as a filter in the dashboard. The issue with using the `object` type is that the tags can't be leveraged for filtering since we don't know the mapping beforehand.

Reply:
> My concern with this implementation is that we no longer have the original event which aggregates `total_exact`. Instead, we now have all of the events that add up to `total_exact`.

I am not clear on this @lucian-ioan .
The total exact will be calculated per row after the grouping. And I think that is the right thing to do. 
Initially it was a sum of all the total_exact's we are now getting in different rows as ween [here](https://github.com/elastic/beats/pull/36902#issuecomment-1775621855).

Reply:
@gpop63, in this PR, it is essential to ensure all the existing features continue to work while we add new fields. 

It's okay to add new visualizations later in a dedicated PR.